enterprise/reporting/event_stats.go

```diff
@@ -0,0 +1,52 @@
+package reporting
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+const (
+	EventsDeliveredMetricName = "events_delivered_total"
+	EventsAbortedMetricName   = "events_aborted_total"
+)
+
+type EventStatsReporter struct {
+	stats            stats.Stats
+	configSubscriber *configSubscriber
+}
+
+func NewEventStatsReporter(configSubscriber *configSubscriber, stats stats.Stats) *EventStatsReporter {
+	return &EventStatsReporter{
+		stats:            stats,
+		configSubscriber: configSubscriber,
+	}
+}
+
+func (es *EventStatsReporter) Report(metrics []*types.PUReportedMetric) error {
+	// tracking delivery event stats which has state - succeeded, aborted
+	for _, metric := range metrics {
+		metric := metric
```

Comment:
Not sure why we need to create a local shadow here?
```suggestion
		metric := metric
```

Reply:
right not needed. if there is a go routine call in for loop. Then it will cause an issue